sdk/keyvault/Azure.Security.KeyVault.Keys/tests/KeyClientTests.cs

```diff
@@ -222,6 +223,42 @@ public void ReleaseKeyParameterValidation()
             Assert.AreEqual("target", ex.ParamName);
         }
 
+        [Test]
+        [SyncOnly]
+        public void GetCryptographyClientValidation()
+        {
+            ArgumentException ex = Assert.Throws<ArgumentNullException>(() => Client.GetCryptographyClient(null));
+            Assert.AreEqual("name", ex.ParamName);
+
+            ex = Assert.Throws<ArgumentException>(() => Client.GetCryptographyClient(string.Empty));
+            Assert.AreEqual("name", ex.ParamName);
+        }
+
+        [Test]
+        public async Task GetCryptographyClientUsesSamePipeline()
+        {
+            // Make sure the created CryptographyClient uses the same mock transport as the KeyVault that created it.
+            MockTransport transport = new(new[]
+            {
+                new MockResponse(200).WithContent(@"{""attributes"":{""created"":1626299777,""enabled"":true,""exportable"":false,""updated"":1626299777},""key"":{""key_ops"":[""wrapKey"",""unwrapKey""],""kid"":""https://test.managedhsm.azure.net/keys/test/abcd1234"",""kty"":""oct-HSM""}}"),
+                new MockResponse(200).WithContent(@"{""alg"":""A128KW"",""kid"":""https://test.managedhsm.azure.net/keys/test/abcd1234"",""value"":""dGVzdA""}"),
+            });
+
+            KeyClientOptions options = new()
+            {
+                Transport = transport,
+            };
+
+            KeyClient keyClient = new(new Uri("https://localhost"), new MockCredential(), options);
+            KeyVaultKey key = await keyClient.CreateEcKeyAsync(new CreateEcKeyOptions("test"));
+            Assert.AreEqual("test", key.Name);
+            Assert.AreEqual("abcd1234", key.Properties.Version);
+
+            CryptographyClient cryptographyClient = keyClient.GetCryptographyClient(key.Name, key.Properties.Version);
```

Comment:
Today you can new up a crypto client by passing in a keyId or JWK (for local-only which we can ignore in this case since it's a different enough use-case). By using name+version here, we're introducing a third way to instantiate a cryptography client. I wonder if this should take in a `Uri keyId` instead of building it up from name and version for consistency?

Reply:
They can just create a new `CryptographyClient` then. All our methods on the `KeyClient` take a name and optionally a version (or not). This is a consistent pattern with the `KeyClient` since that's where this method is defined. Why make them pass a full `Uri` where the endpoint might actually be different from the `KeyClient`? That's just confusing.

This is what I expect of all languages' implementation: we match the pattern of the `KeyClient` methods because this is another method on the `KeyClient`. It also fits the .NET guidelines I was aiming for.

Reply:
Fair enough, thanks for the reply!